src/libraries/System.Data.Common/src/System/Data/DataTableReader.cs

```diff
@@ -130,7 +130,7 @@ public override void Close()
             _isOpen = false;
         }
 
-        public override DataTable GetSchemaTable()
+        public override DataTable? GetSchemaTable()
```

Comment:
This method never seems to actually return null (even if GetSchemaTable in DbDataReader does), so we should probably annotate this as non-nullable.

Reply:
Yes, it was not returning null, and `?` was removed from ref, but reverted by review comment https://github.com/dotnet/runtime/pull/41731#discussion_r482841465, looked it again seems you are right, not returning even parent is nullable cc @stephentoub

Reply:
This one has gone back and forth and it's unclear to me what we're concluding is the final decision here. @buyaa-n / @roji Do we want to change this again to land on not nullable, or are we keeping it as nullable?

Reply:
As derived type may narrow the return type from `T?` to `T` and @roji confirmed it should be non null I am gonna update this to `T`

Reply:
if we're not sure we should annotate it as nullable so that it would not be a breaking change to change it back in vNext

Reply:
@buyaa-n I think that's the right thing to do.

For context, DbDataReader.GetSchemaTable is indeed nullable, and null is returned when a database command doesn't return any resultset at all (or all resultsets have already been consumed on the reader via NextResult). This is how all ADO.NET providers behave (SqlClient, Odbc, OleDb...), see discussion in #41082 and #509.

DataTableReader is a specific implementation that exposes data in an existing DataTable via a DbDataReader. The implementation in this class never returns null, as a DataTable always provides an existing "resultset", so code working against DataTableReader can safely assume null will never be returned.